### PR TITLE
cmake: require CMake v3.16 and cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-cmake-cleanups-kefu/9a8a5cc3fdbbc87a4974245ffa87403763072896/

File: CMakeLists.txt [hunk 1]
```diff
@@ -1,25 +1,19 @@
-cmake_minimum_required(VERSION 3.10.2)
-# remove cmake/modules/FindPython* once 3.12 is required
```
Comment (ideepika) [inline CMakeLists.txt:2]: @tchaikov do we plan on removing these as well, if that is still remaining work, can we keep a note of it somewhere

**Review by ideepika — APPROVED** [review]
thanks @tchaikov ! +1

**Comment by tchaikov** [comment]
changelog:

- avoid option de-duplication by adding `SHELL:`.

**Comment by tchaikov** [comment]
https://shaman.ceph.com/builds/ceph/wip-cmake-cleanups-kefu/0270f73e6ef3c9ac72e9f739ef0aaadfb87e49b9/

**Pull request merged**